### PR TITLE
fix(dropdown): add automatic resize listener

### PR DESCRIPTION
**Pull request by SamCreasey**
Added a resize event to automatically recalculate divider height

### CODE REVIEW

File: src/components/Dropdown/Dropdown.tsx [hunk 1]
```diff
@@ -141,7 +164,7 @@ export const Dropdown: FC<DropdownProps> = ({
                 </button>
             </Trigger>
             <AnimatePresence>
-                {!disabled && isOpen && (
+                {!disabled && isOpen && maxHeight !== DEFAULT_DROPDOWN_MAX_HEIGHT && (
```
Comment (SamuelAlev) [inline src/components/Dropdown/Dropdown.tsx:167]: Why is it needed?

File: src/components/Dropdown/Dropdown.tsx [hunk 2]
```diff
@@ -154,9 +177,10 @@ export const Dropdown: FC<DropdownProps> = ({
                             <div
                                 {...overlayProps}
                                 ref={overlayRef}
-                                style={{ maxHeight: getInnerOverlayHeight(triggerRef) }}
+                                style={{ maxHeight }}
```
Comment (SamuelAlev) [inline src/components/Dropdown/Dropdown.tsx:180]:
```suggestion
                                style={autoResize ? { maxHeight } : {}}
```

File: src/components/Dropdown/Dropdown.tsx [hunk 3]
```diff
@@ -99,6 +103,25 @@ export const Dropdown: FC<DropdownProps> = ({
...
+            if (autoResize) {
+                window.addEventListener("resize", updateMaxHeight);
+            }
+        } else {
+            setMaxHeight(DEFAULT_DROPDOWN_MAX_HEIGHT);
+        }
+        return () => {
+            if (isOpen && autoResize) {
+                window.removeEventListener("resize", updateMaxHeight);
+            }
```
Comment (SamuelAlev) [inline src/components/Dropdown/Dropdown.tsx:121]:
```suggestion
            window.removeEventListener("resize", updateMaxHeight);
```

File: src/components/Dropdown/Dropdown.tsx [hunk 4]
```diff
@@ -99,6 +103,25 @@ export const Dropdown: FC<DropdownProps> = ({
...
+    useEffect(() => {
+        const updateMaxHeight = () => setMaxHeight(getInnerOverlayHeight(triggerRef));
+        if (isOpen) {
+            updateMaxHeight();
+            if (autoResize) {
+                window.addEventListener("resize", updateMaxHeight);
+            }
+        } else {
+            setMaxHeight(DEFAULT_DROPDOWN_MAX_HEIGHT);
+        }
```
Comment (SamuelAlev) [inline src/components/Dropdown/Dropdown.tsx:117]: Basically we can omit the whole max height logic if  `autoResize === false`

File: src/components/Dropdown/Dropdown.tsx (hunk 1 above)
Comment (SamCreasey) [reply src/components/Dropdown/Dropdown.tsx:167]: It basically prevents the dropdown from rendering and then the max height changing immediately afterwards causing a small visual bug

File: src/components/Dropdown/Dropdown.tsx (hunk 4 above)
Comment (SamCreasey) [reply src/components/Dropdown/Dropdown.tsx:117]: Ah I see that's what you meant, I thought you meant just for the window listener

File: src/components/Dropdown/Dropdown.tsx (hunk 2 above)
Comment (SamCreasey) [reply src/components/Dropdown/Dropdown.tsx:180]: fixed

File: src/components/Dropdown/Dropdown.tsx (hunk 3 above)
Comment (SamCreasey) [reply src/components/Dropdown/Dropdown.tsx:121]: fixed

**Review by Walkeezy — APPROVED** [review]

**Pull request merged**